rmf_site_format/src/scenario.rs

```diff
@@ -19,13 +19,78 @@ use crate::*;
 #[cfg(feature = "bevy")]
 use bevy::prelude::{Bundle, Component, Reflect, ReflectComponent};
 use serde::{Deserialize, Serialize};
-use std::collections::HashMap;
+use std::collections::{BTreeMap, HashMap};
 
 #[derive(Serialize, Deserialize, Debug, Clone, Copy, Default, PartialEq, Eq)]
 #[cfg_attr(feature = "bevy", derive(Component, Reflect))]
 #[cfg_attr(feature = "bevy", reflect(Component))]
 pub struct InstanceMarker;
 
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "bevy", derive(Component))]
+pub enum InstanceModifier {
+    Added(AddedInstance),
+    Inherited(InheritedInstance),
+    Hidden,
+}
+
+impl InstanceModifier {
+    pub fn new_added(pose: Pose) -> Self {
+        Self::Added(AddedInstance { pose: pose })
+    }
+
+    pub fn new_inherited(pose: Option<Pose>) -> Self {
+        Self::Inherited(InheritedInstance {
+            modified_pose: pose,
+        })
+    }
+
+    pub fn new_hidden() -> Self {
```

Comment:
We could change this to `pub fn hidden()` to match the others, but I think we should just call `InstanceModifier::Hidden` directly.